### PR TITLE
guide-pv-for-baremetal

### PR DESCRIPTION
**Pull request by calvix**
I should not create text for customers as my writing skills are equal to zero.

Maybe someone wanna polish it? 📦 
towards https://github.com/giantswarm/giantswarm/issues/1846

### CODE REVIEW

**Comment by oponder** [comment]
Thanks! I'll take a moment to polish it up a bit.

Can you explain a bit quickly the context, like what will bring someone to this page? Is it an answer to a question? I want to add a little bit more introductory text to explain to the reader what this page is about, so they can see if it applies to them or not.

**Comment by oponder** [comment]
Here are my suggested changes: https://github.com/giantswarm/docs-content/pull/259/files

Let me know if you think this still meets the spirit of what you wanted to say.

**Comment by calvix** [comment]
> Can you explain a bit quickly the context, like what will bring someone to this page? Is it an answer to a question? I want to add a little bit more introductory text to explain to the reader what this page is about, so they can see if it applies to them or not.

 We dont wanna handle guest cluster storage atm, and on most of our on-prem installation they have NFS or iSCSI so we wanted to create guide that we can send customer  so he can use persistent storage on guest cluster

**Comment by oponder** [comment]
So is this something we do for them, or is this like a guide / recommendation of how they can do it themselves?

Do we create those folders and PV's, or do they?

**Comment by calvix** [comment]
its recommendation  how they can do it themselfs

**Comment by calvix** [comment]
can you approve?

File: content/guides/using-persistent-volumes-on-baremetal/index.md [hunk 1]
```diff
@@ -0,0 +1,127 @@
++++
+title = "Using Persistent Volumes on baremetal"
```
Comment (puja108) [inline content/guides/using-persistent-volumes-on-baremetal/index.md]: here and elsewhere `bare metal`

File: content/guides/using-persistent-volumes-on-baremetal/index.md [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+description = "Tutorial on how to use precreated Persistent Volumes on a cluster running on premise (baremetal)"
```
Comment (puja108) [inline content/guides/using-persistent-volumes-on-baremetal/index.md]: `pre-created`

File: content/guides/using-persistent-volumes-on-baremetal/index.md [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+date = "2017-09-19"
+type = "page"
+weight = 50
+tags = ["tutorial"]
++++
+
+# Using Persistent Volumes on your on-prem baremetal clusters
```
Comment (puja108) [inline content/guides/using-persistent-volumes-on-baremetal/index.md]: `on-premise`

File: content/guides/using-persistent-volumes-on-baremetal/index.md [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+
+Persistent Volumes allow you to store data beyond the lifetime of a Pod. Ideally, storage is dynamically provisioned and claiming storage for your pods would not require you to pre-provision storage. However, depending on your installation, your baremetal clusters might not support dynamic storage provisioning.
+
+This document describes how we still allow you to claim volumes for your Pods by pre-creating a known folder structure in either a NFS share or in iSCIS (iSCIS docs coming soon).
```
Comment (puja108) [inline content/guides/using-persistent-volumes-on-baremetal/index.md]: `iSCSI`

File: content/guides/using-persistent-volumes-on-baremetal/index.md [hunk 5]
```diff
@@ -0,0 +1,127 @@
...
+```
+/k8s-storage/pv0001
+/k8s-storage/pv0002
+/k8s-storage/pv0003
+...
+...
+/k8s-storage/pv00xy
+```
+Obviously you can use any naming scheme you want.
+Usually you don't know exact number of volumes you'll need, so it's good idea to preprovison a lot.
```
Comment (puja108) [inline content/guides/using-persistent-volumes-on-baremetal/index.md]: `know exact` -> `know the exact` 

`it's good` -> `it's a good` 

`preprovision` -> `pre-provision`

File: content/guides/using-persistent-volumes-on-baremetal/index.md [hunk 6]
```diff
@@ -0,0 +1,127 @@
...
+  persistentVolumeReclaimPolicy: Recycle
+  storageClassName: slow
+  nfs:
+    path: /k8s-storage/pv0002
+    server: nfs.storageserver.internal
+```
+
+The only values you need to change on each additional Persistent Volume  is `name` and `path`.
+
+Note that the NFS protocol can't enforce storage capacity, however defining a capacity here is needed to match the Persistent Volume Claim with a Persistent Volume.
```
Comment (puja108) [inline content/guides/using-persistent-volumes-on-baremetal/index.md]: `, however` -> `, however,`

File: content/guides/using-persistent-volumes-on-baremetal/index.md [hunk 7]
```diff
@@ -0,0 +1,127 @@
...
+
+## Using Persistent Volumes in a Pod
+
+Once you have a Persistent Volume you can [claim it as a Volume](https://kubernetes.io/docs/concepts/storage/persistent-volumes/#claims-as-volumes) in your Pods.
+
+Note that these Volumes can only be used by a single Pod at the same time. Thus, the access mode of your claim must be `ReadWriteOnce`.
+
+Under the hood the Volume stays detached from your nodes as long as it is not claimed by a Pod. As soon as a Pod claims it, it gets attached to the node that holds the Pod.
+
```
Comment (puja108) [inline content/guides/using-persistent-volumes-on-baremetal/index.md]: Maybe add here something about security:

```
If you want to limit access to certain persistent volumes to a certain namespace, you need to also pre-create the PVCs, as those, unlike PVs, are namespaced and thus allow for access controls on a namespace level.
```

**Comment by calvix** [comment]
@puja108  RFR

**Comment by puja108** [comment]
btw, after merge you need to roll out via empty commit on `docs` repo currently

File: content/guides/using-persistent-volumes-on-baremetal/index.md [hunk 8]
```diff
@@ -0,0 +1,129 @@
...
+kind: PersistentVolumeClaim
+apiVersion: v1
+metadata:
+  name: myclaim
+spec:
+  accessModes:
+    - ReadWriteOnce
+  resources:
+    requests:
+      storage: 6Gi
```
Comment (puja108) [inline content/guides/using-persistent-volumes-on-baremetal/index.md]: Maximilian from CS mentioned that this should fail as above we define 5Gi for the PV. ping @calvix

**Pull request merged**